lib/task/export/csvExportTermUsageTask.class.php

```diff
@@ -0,0 +1,139 @@
+    ));
+    $this->addOptions(array(
+      new sfCommandOption('taxonomy-name', null, sfCommandOption::PARAMETER_OPTIONAL, 'Name of taxonomy')
+    ));
+    $this->addOptions(array(
+      new sfCommandOption('taxonomy-name-culture', null, sfCommandOption::PARAMETER_OPTIONAL, 'Culture to use for taxonomy name lookup')
+    ));
+    $this->addOptions(array(
+      new sfCommandOption('rows-per-file', null, sfCommandOption::PARAMETER_OPTIONAL, 'Rows per file (disregarded if writing to a file, not a directory)', false)
+    ));
+  }
+
+  /**
+   * @see sfTask
+   */
+  public function execute($arguments = array(), $options = array())
+  {
+    if (isset($options['items-until-update']) && !ctype_digit($options['items-until-update']))
+    {
+      throw new sfException('items-until-update must be a number');
+    }
+
+    $configuration = ProjectConfiguration::getApplicationConfiguration('qubit', 'cli', false);
+    $sf_context = sfContext::createInstance($configuration);
+    $conn = $this->getDatabaseConnection();
+
+    $itemsExported = $this->exportToCsv($this->determineTaxonomyId($options), $arguments['path'], $options['items-until-update']);
+    $this->log(sprintf("\nExport complete (%d terms exported).", $itemsExported));
+  }
+
+  private function determineTaxonomyId($options)
+  {
+    if (ctype_digit($options['taxonomy-id']))
+    {
+      $criteria = new Criteria;
+      $criteria->add(QubitTaxonomy::ID, $options['taxonomy-id']);
+
+      if (null === QubitTaxonomy::getOne($criteria))
+      {
+        throw new sfException('Invalid taxonomy-id.');
+      }
+
+      return $options['taxonomy-id'];
+    }
+    else if(isset($options['taxonomy-name']))
+    {
+      $culture = (isset($options['taxonomy-name-culture'])) ? $options['taxonomy-name-culture'] : 'en';
+
+      $criteria = new Criteria;
+
+      $criteria->add(QubitTaxonomyI18n::NAME, $options['taxonomy-name']);
+      $criteria->add(QubitTaxonomyI18n::CULTURE, $culture);
+
+      if (null === $taxonomy = QubitTaxonomyI18n::getOne($criteria))
+      {
+        throw new sfException('Invalid taxonomy-name and/or taxonomy-name-culture.');
+      }
+
+      return $taxonomy->id;
+    }
+
+    throw new sfException('Either the taxonomy-id or taxonomy-name must be used to specifiy a taxonomy.');
+  }
+
+  private function exportToCsv($taxonomyId, $exportPath, $rowsUntilUpdate)
+  {
+    $itemsExported = 0;
+
+    /*
+
+    Idea for future:
+
+    1. Count uses with SELECT DISTINCT t.id, COUNT(object_id) AS uses, r.object_id FROM term t INNER JOIN object_term_relation r ON r.term_id=t.id WHERE t.taxonomy_id=35 GROUP BY(id) ORDER BY t.id;
+    2. Add uses to "usage" export config and change QubitEport so setColumn will work before exportResource
+    3. LEFT JOIN could include unused terms so we could count those too
+    4. Add --mode=used|both|unused-only ("used" as default) to flip between LEFT and INNER join (and maybe add criteria for specific modes)
+
+    */
```

Comment:
Could you remove the empty lines at the beginning and end and add one after the comment block? And I prefer the full block ...

/*
 *
 *
 */